qiskit/transpiler/passes/layout/full_ancilla_allocation.py

```diff
@@ -81,7 +81,7 @@ def run(self, dag):
 
         idle_physical_qubits = [q for q in layout_physical_qubits if q not in physical_bits]
 
-        if self.target:
+        if self.target and self.target.num_qubits is not None:
```

Comment:
Not required but explicitly checking for `None` is a bit safer (and would've prevented some bugs in the past 🙂)
```suggestion
        if self.target is not None and self.target.num_qubits is not None:
```